dataactvalidator/validation_handlers/validationManager.py

```diff
@@ -509,30 +544,57 @@ def run_cross_validation(self, job):
                 RuleSql.target_file_id == second_file.id), and_(
                 RuleSql.file_id == second_file.id,
                 RuleSql.target_file_id == first_file.id)))
-            # send comboRules to validator.crossValidate sql
-            failures = cross_validate_sql(combo_rules.all(), submission_id, self.short_to_long_dict, first_file.id,
-                                          second_file.id, job)
-            # get error file name
-            report_filename = self.get_file_name(report_file_name(submission_id, False, first_file.name,
-                                                                  second_file.name))
-            warning_report_filename = self.get_file_name(report_file_name(submission_id, True, first_file.name,
-                                                                          second_file.name))
-
-            # loop through failures to create the error report
-            with self.get_writer(region_name, bucket_name, report_filename, self.crossFileReportHeaders) as writer, \
-                    self.get_writer(region_name, bucket_name, warning_report_filename, self.crossFileReportHeaders) as \
-                    warning_writer:
-                for failure in failures:
-                    if failure[9] == RULE_SEVERITY_DICT['fatal']:
-                        writer.write(failure[0:7])
-                    if failure[9] == RULE_SEVERITY_DICT['warning']:
-                        warning_writer.write(failure[0:7])
-                    error_list.record_row_error(job_id, "cross_file",
-                                                failure[0], failure[3], failure[5], failure[6],
-                                                failure[7], failure[8], severity_id=failure[9])
-                # write the last unfinished batch
-                writer.finish_batch()
-                warning_writer.finish_batch()
+
+            # get error file name/path
+            error_file_name = report_file_name(submission_id, False, first_file.name, second_file.name)
+            error_file_path = "".join([CONFIG_SERVICES['error_report_path'], error_file_name])
+            warning_file_name = report_file_name(submission_id, True, first_file.name, second_file.name)
+            warning_file_path = "".join([CONFIG_SERVICES['error_report_path'], warning_file_name])
+
+        # open error report and gather failed rules within it
+            with open(error_file_path, 'w', newline='') as error_file,\
```

Comment:
This seems like duplicate code. Can we separate this out into its own function?

Reply:
No, because they do very different things within the `with`